CRM/Contact/Form/Relationship.php

```diff
@@ -659,27 +660,6 @@ private function setEmploymentRelationship($params, $relationshipIds) {
       }
       // @todo this belongs in the BAO.
       CRM_Contact_BAO_Contact_Utils::setCurrentEmployer($employerParams);
-      // Refresh contact summary if in ajax mode
-      $this->ajaxResponse['reloadBlocks'] = array('#crm-contactinfo-content');
-    }
-  }
-
-  /**
-   * Clears the current employer if the relationship type
-   * get changed, disabled or 'current employer' checkbox get unchecked.
-   *
-   * @param $params
-   */
-  private function clearCurrentEmployer($params) {
-    // @todo this belongs in the BAO.
-    if ($this->_isCurrentEmployer) {
-      $relChanged = $params['relationship_type_id'] != $this->_values['relationship_type_id'];
-      if (!$params['is_active'] || !$params['is_current_employer'] || $relChanged) {
```

Comment:
the $params['is_current_employer'] functionality does seem to be form-relevant - ie there could be 2 employers & one (or possibly neither?) could be form-selected - I do like the move to bao for rel type change tho

Reply:
Tested on assigning relationship through search results and we don't offer `current employer` checkbox on that form. I found another issue(before this PR) which failed to remove current employer when the checkbox is unchecked while editing the relationship. It has been fixed in the changes made here.

Reply:
@jitendrapurohit just spotted one more thing - looks like the original code would also clear current employer if toggled to inactive (presumably date changes could trigger that too but I think that could stay out of scope since it was not part of the previous behaviour)

Reply:
Thanks, this brings another related issue on dmaster which fails to reset current employer after `is_active` checkbox is unchecked and the form is saved. Added fix for the same to this PR.